app/src/main/java/org/mozilla/firefox/vpn/apptunneling/data/AppTunnelingRepository.kt

```diff
@@ -0,0 +1,88 @@
+package org.mozilla.firefox.vpn.apptunneling.data
+
+import android.content.SharedPreferences
+import android.content.pm.ApplicationInfo
+import android.content.pm.PackageManager
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import org.mozilla.firefox.vpn.BuildConfig
+import org.mozilla.firefox.vpn.util.putStringSetSafe
+
+class AppTunnelingRepository(
+    private val packageManager: PackageManager,
+    private val sharedPreferences: SharedPreferences
+) {
+
+    companion object {
+        const val PREF_KEY_STRINGSET_EXCLUDE = "key_stringset_apptunneling_exclude_list"
+        const val PREF_KEY_USE_APP_TUNNELING = "key_use_app_tunneling"
+    }
+
+    fun getPackages(includeInternalApps: Boolean): List<ApplicationInfo> {
+        val applicationInfoList = packageManager.getInstalledApplications(0)
+        val packageList = applicationInfoList.filter {
+            PERMISSION_GRANTED == packageManager.checkPermission(
+                android.Manifest.permission.INTERNET,
+                it.packageName
+            )
+        }.filter {
+            includeInternalApps or ((it.flags and ApplicationInfo.FLAG_SYSTEM) == 0)
```

Comment:
I think this should be
`includeInternalApps **||** ((it.flags and ApplicationInfo.FLAG_SYSTEM) == 0)`
instead of 
`includeInternalApps **or** ((it.flags and ApplicationInfo.FLAG_SYSTEM) == 0)`